### ISSUE TITLE
airspec: org.scalajs.testing.adapter.JSEnvRPC$RunTerminatedException

### ISSUE THREAD

**Issue by xerial** [wvlet/airframe#2893]
airspec runner occasionally fails even for a simple spec. 

Example failure: https://github.com/wvlet/airframe/runs/13077205895

Probably related to https://github.com/scala-js/scala-js/issues/4317

Related code 
- Scala.js: https://github.com/scala-js/scala-js/blob/2e8549c49716218ca1be18ea854c2fa66738d4f6/test-adapter/src/main/scala/org/scalajs/testing/adapter/JSEnvRPC.scala#L59-L60
- scala-js-js-envs: https://github.com/scala-js/scala-js-js-envs/blob/d0683564ef64c86731e1037a30d5bec13c95b5fd/js-envs/src/main/scala/org/scalajs/jsenv/ExternalJSRun.scala#L200
- Scala.js JUnit runner implementation https://github.com/scala-js/scala-js/blob/main/junit-runtime/src/main/scala/org/scalajs/junit/JUnitTask.scala
```
org.scalajs.testing.common.RPCCore$ClosedException: org.scalajs.testing.adapter.JSEnvRPC$RunTerminatedException
	at org.scalajs.testing.common.RPCCore.helpClose(RPCCore.scala:223)
	at org.scalajs.testing.common.RPCCore.call(RPCCore.scala:164)
	at org.scalajs.testing.common.RunMuxRPC.call(RunMuxRPC.scala:43)
	at org.scalajs.testing.adapter.TaskAdapter.execute(TaskAdapter.scala:45)
	at sbt.TestRunner.runTest$1(TestFramework.scala:147)
	at sbt.TestRunner.run(TestFramework.scala:162)
	at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:330)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:290)
	at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:330)
	at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:330)
	at sbt.TestFunction.apply(TestFramework.scala:342)
	at sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
	at sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
	at sbt.Execute.work(Execute.scala:291)
	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.scalajs.testing.adapter.JSEnvRPC$RunTerminatedException
	at org.scalajs.testing.adapter.JSEnvRPC.$anonfun$new$1(JSEnvRPC.scala:60)
	at org.scalajs.testing.adapter.JSEnvRPC.$anonfun$new$1$adapted(JSEnvRPC.scala:60)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: org.scalajs.jsenv.ExternalJSRun$NonZeroExitException: exited with code 1
	at org.scalajs.jsenv.ExternalJSRun$$anon$1.run(ExternalJSRun.scala:200)
```

More errors:
```
JSRPCClientTest:
 - Create an Async RPCClient
2023-04-27T18:56:31.482Z  info [JSRPCClientTest] TestResponse(https://httpbin.org/post,HashMap(Referer -> http://localhost/, Origin -> http://localhost, Content-Length -> 22, Accept -> */*, Content-Type -> application/json;charset=utf-8, Accept-Language -> en, Accept-Encoding -> gzip, deflate, X-Amzn-Trace-Id -> Root=1-644ac55f-38a2e7e96dc81fc95c60adc0, User-Agent -> Mozilla/5.0 (darwin) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/20.0.0, Host -> httpbin.org))  - (JSRPCClientTest.scala:41)
     - rpc 305.19ms
java.lang.IllegalStateException: Promise already completed.
java.lang.IllegalStateException: Promise already completed.
    at $f_s_concurrent_Promise__complete__s_util_Try__s_concurrent_Promise (file:///Users/leo/work/git/airframe/airframe-http/.js/target/scala-2.13/airframe-http-test-fastopt/main.js:15837:45)
    at $f_s_concurrent_Promise__failure__jl_Throwable__s_concurrent_Promise (file:///Users/leo/work/git/airframe/airframe-http/.js/target/scala-2.13/airframe-http-test-fastopt/main.js:15846:10)
    at $c_Lwvlet_airframe_http_client_JSHttpClientChannel.wvlet$airframe$http$client$JSHttpClientChannel$$$anonfun$sendAsync$2__Lorg_scalajs_dom_Event__s_concurrent_Promise__Lwvlet_airframe_http_HttpMessage$Request__s_concurrent_Promise (file:///Users/leo/work/git/airframe/airframe-http/.js/target/scala-2.13/airframe-http-test-fastopt/main.js:107233:10)
    at XMLHttpRequest.$c_Lwvlet_airframe_http_client_JSHttpClientChannel.sendAsync__Lwvlet_airframe_http_HttpMessage$Request__Lwvlet_airframe_http_ChannelConfig__Lwvlet_airframe_rx_Rx.xhr.onerror (file:///Users/leo/work/git/airframe/airframe-http/.js/target/scala-2.13/airframe-http-test-fastopt/main.js:107215:35)
    at XMLHttpRequest.invokeTheCallbackFunction (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/generated/EventHandlerNonNull.js:14:28)
    at XMLHttpRequest.<anonymous> (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/helpers/create-event-accessor.js:35:32)
    at innerInvokeEventListeners (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:350:25)
    at invokeEventListeners (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:286:3)
    at XMLHttpRequestImpl._dispatch (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:233:9)
    at fireAnEvent (/Users/leo/work/git/airframe/node_modules/jsdom/lib/jsdom/living/helpers/events.js:18:36)
[error] stack trace is suppressed; run last httpJS / Test / executeTests for the full output
```

**Comment by xerial** [comment]
I hope this fix will catch such an exception #2895

**Comment by xerial** [comment]
The exception is happening at the sbt test runner, so AirSpec can't fix this.
An example error #2905 

Fixed by using a different free REST service for Scala.js #2906
